src/tests/BuildWasmApps/Wasm.Build.Tests/PInvokeTableGeneratorTests.cs

```diff
@@ -119,6 +119,198 @@ public static int Main()
             Assert.Contains("Main running", output);
         }
 
+        [Theory]
+        [BuildAndRun(host: RunHost.None)]
+        public void UnmanagedStructsAreNotConsideredBlittable(BuildArgs buildArgs, string id)
+        {
+            string code =
+            """
+            using System;
+            using System.Runtime.CompilerServices;
+            using System.Runtime.InteropServices;
+
+            public class Test
+            {
+                public static int Main()
+                {
+                    var x = new S { Value = 5 };
+
+                    Console.WriteLine("Main running " + x.Value);
+                    return 42;
+                }
+
+                public struct S { public int Value; }
+
+                [UnmanagedCallersOnly]
+                public static void M(S myStruct) { }
+            }
+            """;
+
+            buildArgs = ExpandBuildArgs(
+                buildArgs with { ProjectName = $"not_blittable_{buildArgs.Config}_{id}" },
+                extraProperties: buildArgs.AOT
+                    ? string.Empty
+                    : "<WasmBuildNative>true</WasmBuildNative>"
+            );
+
+            (_, string output) = BuildProject(
+                buildArgs,
+                id: id,
+                new BuildProjectOptions(
+                    InitProject: () =>
+                    {
+                        File.WriteAllText(Path.Combine(_projectDir!, "Program.cs"), code);
+                    },
+                    Publish: buildArgs.AOT,
+                    DotnetWasmFromRuntimePack: false,
+                    ExpectSuccess: false
+                )
+            );
+
+            Assert.Matches("error.*Parameter.*types.*pinvoke.*.*blittable", output);
+        }
+
+        [Theory]
+        [BuildAndRun(host: RunHost.Chrome)]
+        public void UnmanagedStructsAreConsideredBlittable(BuildArgs buildArgs, RunHost host, string id)
```

Comment:
The name should reflect what case this is, since there are multiple here.

The cases that should be covered IMHO, are:

1. Method, and struct in same assembly with no attribute
2. Separate assemblies - only Struct one has attribute
3. Separate assemblies - only method one has attribute
4. Separate assemblies - both have attribute
5. Separate assemblies - neither have the attribute

They are very similar, so having the test as a method which gets used by different cases, should make it easier to understand too.

Reply:
I'll add them.